src/endpoints/get_expiring_domains.rs

```diff
@@ -0,0 +1,75 @@
+use crate::{
+    models::AppState,
+    utils::get_error,
+};
+use axum::{
+    extract::State,
+    http::{HeaderMap, HeaderValue, StatusCode},
+    response::{IntoResponse, Json},
+};
+use axum_auto_routes::route;
+use futures::StreamExt;
+use mongodb::bson::{doc, Document};
+use std::sync::Arc;
+use serde::Serialize;
+
+#[derive(Serialize)]
+struct IdDetails {
+    addr: String,
+    domain: String,
+}
+
+#[derive(Serialize)]
+pub struct ExpiringDomains {
+    ids: Vec<IdDetails>,
+}
+
+#[route(get, "/get_expiring_domains", crate::endpoints::get_expiring_domains)]
+pub async fn handler(
+    State(state): State<Arc<AppState>>,
+) -> impl IntoResponse {
+    let mut headers = HeaderMap::new();
+    headers.insert("Cache-Control", HeaderValue::from_static("max-age=30"));
+
+    let collection = state.starknetid_db.collection::<Document>("domains");
+
+    let current_time = chrono::Utc::now().timestamp();
+    let one_week_later = current_time + 604800; // Add one week in seconds
+
+    // Define the aggregation pipeline
+    let pipeline = vec![
+        doc! {
+            "$match": {
+                "expiry": { "$lt": one_week_later },
```

Comment:
Please also check that the domain expiry is greater than current_time, because the route is currenlty also returning expired domain